File: submission/solution-mohith-krishna/tests/helpers/builders.ts
import { v4 as uuidv4 } from 'uuid';

export interface TransferPayload {
  source_wallet_id: string;
  destination_wallet_id: string;
  amount: number;
  currency: string;
  reference?: string;
}

export function buildTransferPayload(overrides: Partial<TransferPayload> = {}): TransferPayload {
  return {
    source_wallet_id: 'wallet_001',
    destination_wallet_id: 'wallet_002',
    amount: 2500,
    currency: 'AED',
    reference: 'invoice_123',
    ...overrides,
  };
}

export function buildIdempotencyKey(): string {
  return uuidv4();
}

File: submission/neha/tests/helpers/builders.py
"""Request builders that keep scenario setup readable."""

from __future__ import annotations

import uuid
from typing import Any


def transfer_payload(
    *,
    source: str = "wallet_001",
    destination: str = "wallet_002",
    amount: int = 1000,
    currency: str = "AED",
    reference: str | None = "invoice_123",
    **overrides: Any,
) -> dict[str, Any]:
    payload: dict[str, Any] = {
        "source_wallet_id": source,
        "destination_wallet_id": destination,
        "amount": amount,
        "currency": currency,
    }
    if reference is not None:
        payload["reference"] = reference
    payload.update(overrides)
    return payload


def new_idempotency_key(prefix: str = "key") -> str:
    return f"{prefix}-{uuid.uuid4()}"
